Sustainability/OCP-Sustainability-Profile.json

```diff
@@ -2,37 +2,91 @@
+                    ],
                     "PropertyRequirements": {
-                        "Reading": {}
+                        "PowerWatts": {
+                            "ReadRequirement": "Supported",
+                            "Purpose": "Notate the sampling rate, etc.",
+                            "PropertyRequirements": {
+                                "Reading": {}
+                            }
+                        },
+                        "EnergykWh": {
+                            "ReadRequirement": "Supported",
+                            "Purpose": "Notate the sampling rate, accuracy, etc.",
+                            "PropertyRequirements": {
+                                "Reading": {},
+                                "LifetimeReading": {
+                                    "ReadRequirement": "Recommended"
+                                }
+                            }
+                        }
                     }
                 },
-                "EnergykWh": {
-                    "ReadRequirement": "Supported",
-                    "Purpose": "Notate the sampling rate, accuracy, etc.",
+                {
+                    "UseCaseTitle": "Processor and Memory",
+                    "UseCaseType": "Normal",
+                    "ReadRequirement": "Recommended",
+                    "Purpose": "It is recommended for products with Processor or Memory resources to provide power and energy consumption values for each device.",
+                    "URIs": [
+                        "/redfish/v1/Systems/{ComputerSystemId}/Processors/{ProcessorId}/EnvironmentMetrics",
+                        "/redfish/v1/Systems/{ComputerSystemId}/Memory/{MemoryId}/EnvironmentMetrics"						
+                    ],
+                    "PropertyRequirements": {
+                        "PowerWatts": {
+                            "ReadRequirement": "Recommended",
+                            "PropertyRequirements": {
+                                "Reading": {}
+                            }
+                        },
+                        "EnergykWh": {
+                            "ReadRequirement": "Recommended",
+                            "PropertyRequirements": {
+                                "Reading": {}
+                            }
+                        }
+                    }
+                },
+                {
+                    "UseCaseTitle": "Thermal Subsystem",
+                    "UseCaseType": "Normal",
+                    "ReadRequirement": "Recommended",
+                    "Purpose": "It is recommended for products with fans or other active thermal management components to provide power and energy consumption values for the entire thermal subsystem.  These values are used to calculate energy efficiency.",
+                    "URIs": [
+                        "/redfish/v1/Chassis/{ChassisId}/ThermalSubsystem/EnvironmentMetrics"						
```

Comment:
![image](https://github.com/opencomputeproject/HWMgmt-OCP-Profiles/assets/2031627/8f62ccf8-78c6-4429-afb7-6b171db73f78)

Source: https://www.researchgate.net/figure/Component-wise-energy-consumption-of-a-server-23-24_fig5_355862079

For PUE calculation purposes, a distinction needs to be made between energy spent on "IT" versus "non-IT" (fans and cooling being an example of the latter).

Since CPU and memory only represent 32% and 14% respectively, how do we propose to handle the other categories of IT related energy usage? Shouldn't the profile have a way to represent 100% of energy usage (even if some implementations might report '0' or not include certain elements)?

Reply:
This initial version of the profile doesn't fully contain requirements for calculating PUE-style efficiency within a product.  I am working with the DCEM workstream to define the additional requirements to ensure those metrics can be obtained.  

As the "ThermalSubsystem" power reporting is a new measurement point (not generally implemented in existing hardware or firmware), it's being shown here as a recommendation to show the direction we expect to go in the future.  What is not covered by the profile are similar recommendations to report power conversion losses.  That information is available in the Redfish schema (the PowerSupplyMetrics has properties for reporting both input and output power measurements - with the difference being the conversion losses) which covers both PSU and VRM devices (called "DCRegulator").  It would certainly make sense to add those items as recommendations here as well.  I will take that up with the group to add that to the profile.  
The doesprovide for the PUE to be calculated - if all of the recommended measurement points are implemented by the device.  The profile requires the "Chassis" level power/energy reporting, which will be the "Total" (100%) power consumed.  The "ThermalSubsystem" power (the line this comment is on) reporting reports the losses from all cooling fans or related components.